lib/improver/tests/ensemble_calibration/ensemble_calibration/test_EstimateCoefficientsForEnsembleCalibration.py

```diff
@@ -591,6 +629,59 @@ def test_realizations_predictor_estimate_coefficients(self):
             no_of_realizations=self.no_of_realizations)
         self.assertArrayAlmostEqual(result, data)
 
+    @ManageWarnings(
+        ignored_messages=IGNORED_MESSAGES, warning_types=WARNING_TYPES)
+    def test_mean_predictor_estimate_coefficients_masked_halo(self):
+        """
+        Test that the plugin returns the expected values for the initial guess
+        for the calibration coefficients, when the ensemble mean is used
+        as the predictor. The coefficients are estimated using a linear model,
+        where there is an offset of one between the truth and the forecast
+        during the training period. Therefore, in this case the result of the
+        linear regression is a gradient of 1 and an intercept of 1. In this
+        case the original data has been surrounded by a halo of masked nans,
+        which gives the same coefficients as the original data.
+        """
+        data = np.array([0., 1., 1., 1.], dtype=np.float32)
```

Comment:
Do you think that there would be any value in pulling all the expected results from `Test_compute_initial_guess` into the `setUp`, so that the expected results could be seen side-by-side? This might make things a little clearer.

Reply:
done